tests/Cache/FileHandlerTest.php

```diff
@@ -97,13 +97,16 @@ public function testReadReturnsCache(): void
         self::assertTrue($cached->getSignature()->equals($signature));
     }
 
+    /**
+     * @requires OS Linux|Darwin
+     */
     public function testWriteThrowsIOExceptionIfFileCanNotBeWritten(): void
     {
-        $file = __DIR__.'/non-existent-directory/.php-cs-fixer.cache';
+        $file = '/../../out/of/range/cache.json'; // impossible path
```

Comment:
```suggestion
        $file = __DIR__.str_repeat('/..', 32).'/out/of/range/cache.json'; // impossible path
```
Wouldn't that remove the requirement of `OS Linux|Darwin`? In Unix code is in `/home/runner/work/PHP-CS-Fixer/PHP-CS-Fixer` and this is relative path, so it was trying to create a directory with permission denied. For Windows code is in `D:/a/PHP-CS-Fixer/PHP-CS-Fixer`, so maybe it was able to create that path?

Reply:
It is absolute path, but it's impossible to go above `/`. I believe on Windows it just ignores `..` and converts it into relative path - it's a guess only because I didn't want to play around with Windows 😅. It would be better if some file system abstraction was used, so it would be possible to mock behavior, but with current implementation I just wanted to satisfy condition and test that path at least on Unix.

Reply:
You can never tell with Windows... That's why I suggested 32 `..` to check if that helps.

Reply:
[It still fails](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/actions/runs/5435820544/jobs/9885227238?pr=7109). Personally I would skip this test on Windows, it's not critical 😉. But I did one more test with forbidden char (`"`) in the path.

Reply:
Looks like it works, thanks @kubawerlos for encouraging to find solution here instead of a workaround 🍻.